arm9/source/game/ticket.c

```diff
@@ -28,17 +28,26 @@ u32 ValidateTwlTicket(Ticket* ticket) {
 }
 
 u32 ValidateTicketSignature(Ticket* ticket) {
-    static bool got_modexp = false;
-    static u32 mod[0x100 / 0x4] = { 0 };
-    static u32 exp = 0;
-
-    if (!got_modexp) {
-        // grab mod/exp from cert from cert.db
-        if (LoadCertFromCertDb(0x3F10, NULL, mod, &exp) == 0)
-            got_modexp = true;
-        else return 1;
+    Certificate cert;
```

Comment:
Take note that the previous version of this function stored the mod and exp, so it wouldn't have to go through `certs.db` every time. Performance is important here, so we should keep the mod and exp once we got it. Same applies to `tmd.c

Reply:
Indeed, however as i pointed out on my initial PR comment, only still up for general testing functionality, in regards to mod and exp saving, reasons i removed it for now is that sometimes, a console may have retail and devkit tickets, tmds and or certificates mixed in. And I wonder if i should provide a static storage for the common retail/dev certificates, and fetch from that storage on cert.c if requested again after their first load.

Reply:
Yeah, didn't think about that. Storing mod and exp is important here, though. Maybe store the type (uninitialized, dev, retail) of stored mod and exp here, and only reload if it changes?

Reply:
I can do a static storage with slots for expected common entries, and give them when they were already loaded before on a `Certificate` object, and make Cleanup call check if it is from static storage to not call free on the wrong addresses.

Reply:
Commit aec3989 brings less loading by storing certs whenever they load. My `ticket.db` took about 14 seconds to sort tickets before, now takes about 4 seconds. But still can improve loading modulus with extra functions for `Certificate`.

Reply:
Commit 78c2bc4 adds signature verification functions against certificate, avoiding also copying out modulus, at least when unneeded.